### PR TITLE
[Merged by Bors] - feat: renaming symmetric MvPolynomials

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
Co-authored-by: Junyan Xu <junyanxumath@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/MvPolynomial/Symmetric.lean [hunk 1]
```diff
@@ -157,6 +162,15 @@ end CommRing
 
 end IsSymmetric
 
+/-- `MvPolynomial.rename` induces an isomorphism between the symmetric subalgebras. -/
+def rename_symmetricSubalgebra [CommSemiring R] (e : σ ≃ τ) :
```
Comment (eric-wieser) [inline Mathlib/RingTheory/MvPolynomial/Symmetric.lean:166]:
```suggestion
@[simps!]
def renameSymmetricSubalgebra [CommSemiring R] (e : σ ≃ τ) :
```

File: Mathlib/RingTheory/MvPolynomial/Symmetric.lean [hunk 2]
```diff
@@ -139,6 +139,11 @@ theorem map (hφ : IsSymmetric φ) (f : R →+* S) : IsSymmetric (map f φ) := f
   rw [← map_rename, hφ]
 #align mv_polynomial.is_symmetric.map MvPolynomial.IsSymmetric.map
 
+theorem rename (hφ : φ.IsSymmetric) (e : σ ≃ τ) : (rename e φ).IsSymmetric := fun _ => by
```
Comment (eric-wieser) [inline Mathlib/RingTheory/MvPolynomial/Symmetric.lean:142]: Can you add the `iff` version too, as a `simp` lemma?

File: Mathlib/RingTheory/MvPolynomial/Symmetric.lean [hunk 3]
```diff
@@ -157,6 +162,15 @@ end CommRing
...
 end IsSymmetric
 
+/-- `MvPolynomial.rename` induces an isomorphism between the symmetric subalgebras. -/
+def rename_symmetricSubalgebra [CommSemiring R] (e : σ ≃ τ) :
+    symmetricSubalgebra σ R ≃ₐ[R] symmetricSubalgebra τ R :=
+  AlgEquiv.ofAlgHom
+    (((rename e).comp (symmetricSubalgebra σ R).val).codRestrict _ <| fun x => x.2.rename e)
+    (((rename e.symm).comp <| Subalgebra.val _).codRestrict _ <| fun x => x.2.rename e.symm)
+    (AlgHom.ext <| fun p => Subtype.ext <| by simp)
+    (AlgHom.ext <| fun p => Subtype.ext <| by simp)
```
[the remark this answers is not in the document]
Comment (Ruben-VandeVelde) [reply Mathlib/RingTheory/MvPolynomial/Symmetric.lean]: Not that I could find quickly

**Comment by mattrobball** [comment]
bors merge

File: Mathlib/RingTheory/MvPolynomial/Symmetric.lean [hunk 4]
```diff
@@ -139,6 +139,15 @@ theorem map (hφ : IsSymmetric φ) (f : R →+* S) : IsSymmetric (map f φ) := f
   rw [← map_rename, hφ]
 #align mv_polynomial.is_symmetric.map MvPolynomial.IsSymmetric.map
 
+theorem rename (hφ : φ.IsSymmetric) (e : σ ≃ τ) : (rename e φ).IsSymmetric := fun _ => by
```
Comment (YaelDillies) [inline Mathlib/RingTheory/MvPolynomial/Symmetric.lean:142]:
```suggestion
protected theorem rename (hφ : φ.IsSymmetric) (e : σ ≃ τ) : (rename e φ).IsSymmetric := fun _ => by
```

File: Mathlib/RingTheory/MvPolynomial/Symmetric.lean [hunk 5]
```diff
@@ -139,6 +139,15 @@ theorem map (hφ : IsSymmetric φ) (f : R →+* S) : IsSymmetric (map f φ) := f
...
+  apply rename_injective _ e.symm.injective
+  simp_rw [rename_rename, ← Equiv.coe_trans, Equiv.self_trans_symm, Equiv.coe_refl, rename_id]
+  rw [hφ]
+
+@[simp]
+theorem rename_iff {e : σ ≃ τ} : (MvPolynomial.rename e φ).IsSymmetric ↔ φ.IsSymmetric :=
```
Comment (YaelDillies) [inline Mathlib/RingTheory/MvPolynomial/Symmetric.lean:148]: Should be `MvPolynomial.isSymmetric_rename`

**Review by YaelDillies — APPROVED** [review]

**Comment by mattrobball** [comment]
bors merge

**Comment by mattrobball** [comment]
bors delegate+

**Pull request closed without merging**